src/wattsi.pas

```diff
@@ -1036,6 +1036,37 @@    TCrossReferences = record
          end
       end
       else
+      if (HasAncestor(Element, nsHTML, eTD) or HasAncestor(Element, nsHTML, eDT)) then
+      begin
+         // For elements we're annotating inside a <td> or <dt>, we append the
+         // annotation directly to the <td> or <dt> as the last child. Otherwise
+         // in cases where we have a long table (e.g., the list of events in the
+         // index) or long <dl> list (e.g., the list of pseudo-classes), all the
+         // annos for everything in that table or list end up being merged into
+         // a single annotation way up at the beginning of the table or list.
+         ClassName := 'mdn-anno wrapped before';
+         Candidate := Element;
+         repeat
+            Candidate := Candidate.ParentNode;
+            if (not (Candidate is TElement)) then exit;
+         until (TElement(Candidate).IsIdentity(nsHTML, eTD) or
+                TElement(Candidate).IsIdentity(nsHTML, eDT));
```

Comment:
Could we do this using the same `while not` structure as on lines 1075-1076?

Reply:
> Could we do this using the same `while not` structure as on lines 1075-1076?

Yes — not sure now why I didn’t do it that way to begin with…